jsonschema/src/keywords/pattern.rs

```diff
@@ -43,35 +41,44 @@ impl Validate for PatternValidator {
         instance_path: &InstancePath,
     ) -> ErrorIterator<'a> {
         if let Value::String(item) = instance {
-            if !self.pattern.is_match(item) {
-                return error(ValidationError::pattern(
-                    instance_path.into(),
-                    instance,
-                    self.original.clone(),
-                ));
+            match self.pattern.is_match(item) {
+                Ok(is_match) => {
+                    if !is_match {
+                        return error(ValidationError::pattern(
+                            instance_path.into(),
+                            instance,
+                            self.original.clone(),
+                        ));
+                    }
+                }
+                Err(e) => {
+                    return error(ValidationError::backtrack_limit(
+                        instance_path.into(),
+                        instance,
+                        e,
+                    ));
+                }
             }
         }
         no_error()
     }
 
     fn is_valid(&self, _: &JSONSchema, instance: &Value) -> bool {
         if let Value::String(item) = instance {
-            if !self.pattern.is_match(item) {
-                return false;
-            }
+            return self.pattern.is_match(item).unwrap_or(false);
         }
         true
     }
 }
 
 impl ToString for PatternValidator {
     fn to_string(&self) -> String {
-        format!("pattern: {}", self.pattern)
+        format!("pattern: {:?}", self.pattern)
```

Comment:
Hm, is it required?

Reply:
These were required because `fancy_regex::Regex` implements `Debug` but not `Display`.

Reply:
Aha, got it :) then it obsoletes some of my other comments